src/connect.tsx

```diff
@@ -1,22 +1,51 @@
 import * as React from 'react';
 import hoistNonReactStatics from 'hoist-non-react-statics';
-import createContext from 'create-react-context';
+import createContext, {
+  ProviderProps,
+  ConsumerProps,
+} from 'create-react-context';
 import { FormikContext } from './types';
 
-export const {
-  Provider: FormikProvider,
-  Consumer: FormikConsumer,
-} = createContext<FormikContext<any>>({} as any);
+export const { Provider, Consumer } = createContext<FormikContext<any> | void>(
+  undefined
+);
+
+export function FormikProvider<Values>(
+  props: ProviderProps<FormikContext<Values>>
+) {
+  return <Provider value={props.value} children={props.children} />;
+}
+
+export function FormikConsumer<Values>(
+  props: ConsumerProps<FormikContext<Values>>
+) {
+  return (
+    <Consumer
+      children={formik => {
+        if (typeof formik === 'undefined') {
+          throw new Error(
```

Comment:
I've found this useful in my own use of the new `context` API, to either throw or warn and return `null` when a context dependent field is used outside of the provider. Thoughts?